chart/templates/deployment.yaml

```diff
@@ -0,0 +1,100 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "s3proxy.fullname" . }}
+  labels:
+    {{- include "s3proxy.labels" . | nindent 4 }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  strategy:
+    type: RollingUpdate
+    rollingUpdate:
+      maxSurge: 2        # how many pods we can add at a time
+      maxUnavailable: 0   
+  selector:
+    matchLabels:
+      {{- include "s3proxy.selectorLabels" . | nindent 6 }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        {{- include "s3proxy.selectorLabels" . | nindent 8 }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "s3proxy.serviceAccountName" . }}
+      securityContext:
+        {{ .Values.podSecurityContext.runAsUser  }}
+        {{ .Values.podSecurityContext.runAsGroup  }}
+        {{ .Values.podSecurityContext.fsGroup  }}
+        {{- if semverCompare ">=1.23-0" .Capabilities.KubeVersion.GitVersion -}}
+        {{ .Values.podSecurityContext.fsGroupChangePolicy  }}
+        {{- end }}
+      containers:       
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: http
+              containerPort: 8000
+              protocol: TCP
+          # Note in general use these would fail tracking request for health probe https://github.com/gaul/s3proxy/issues/427
+          # livenessProbe:
+          #   httpGet:
+          #     path: /
+          #     port: http
+          # readinessProbe:
+          #   httpGet:
+          #     path: /
+          #     port: http
+          env:
+            - name: S3PROXY_ENDPOINT
+              value: http://0.0.0.0:8000
+            - name: S3PROXY_IDENTITY
+              valueFrom:
+                secretKeyRef:
+      {{- if empty .Values.existingSecretName }}
+                  name: "{{ include "s3proxy.fullname" . }}-secret"
```

Comment:
I'm confused here. We're setting the user facing S3PROXY_IDENTITY and S3PROXY_CREDENTIAL to AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY values from the secret file.

This is confusing because it implies parity with the backend secrets and does not use the current defaults (local-identity, local-credential).

Reply:
The current "default" uses a known static secret. The helm chart is generating a random key and secret then saving that to a k8s secret. This is for client access. "Existingsecret" allows the admin to precreate the secret and use that instead. This isn't used on the backend